shared/src/business/entities/NewPetitionerUser.ts

```diff
@@ -0,0 +1,171 @@
+import { JoiValidationConstants } from './JoiValidationConstants';
+import { JoiValidationEntity } from './JoiValidationEntity';
+import joi from 'joi';
+
+export type NewPetitionerUserPasswordValidations = {
+  hasNoLeadingOrTrailingSpace: string;
+  hasOneLowercase: string;
+  hasOneNumber: string;
+  hasOneUppercase: string;
+  hasSpecialCharacterOrSpace: string;
+  isProperLength: string;
+};
+const NewPetitionerUserPasswordValidationErrorMessages = {
+  hasNoLeadingOrTrailingSpace: 'Must not contain leading or trailing space',
+  hasOneLowercase: 'Must contain lower case letter',
+  hasOneNumber: 'Must contain number',
+  hasOneUppercase: 'Must contain upper case letter',
+  hasSpecialCharacterOrSpace: 'Must contain special character or space',
+  isProperLength: 'Must be between 8-99 characters long',
+};
+
+function getDefaultErrors(): NewPetitionerUserPasswordValidations {
+  return {
+    hasNoLeadingOrTrailingSpace: '',
```

Comment:
After the refactoring, should we "invert" these variables? i.e. `hasLeadingOrTrailingSpace`, `hasNoLowerCase`, etc